src/orion/core/worker/trial.py

```diff
@@ -250,7 +250,11 @@ def hash_name(self):
         """
         if not self.params:
             raise ValueError("Cannot distinguish this trial, as 'params' have not been set.")
-        return hashlib.md5(self.params_repr().encode('utf-8')).hexdigest()
+        return hashlib.md5((self.params_repr() + str(self.experiment)).encode('utf-8')).hexdigest()
```

Comment:
<3 Future will have seed as well. Let's raise an issue to start implementing determinism.

Reply:
Do you agree with the use of MD5?

Reply:
At least for now, I do :stuck_out_tongue: stress test tries to see if there will be a problem with this.